src/summarizedexperiment/utils.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional, Sequence, MutableMapping
+from functools import reduce
+import numpy as np
+import pandas as pd
+import scipy.sparse as sp
+
+from .SummarizedExperiment import SummarizedExperiment
+
+
+def concat(
```

Comment:
I wouldn't make a separate function, I expect this to be a method of `BaseSE` class, similar to what I did in [GenomicRanges](https://github.com/BiocPy/GenomicRanges/blob/master/src/genomicranges/GenomicRanges.py#L2069)